File: backend/internal/entities/project.go
package entities

type Project struct {
	Id           string        `gorm:"primaryKey" json:"id"`
	Name         string        `gorm:"not null" json:"name"`
	Description  string        `json:"description"`
	UserId       string        `gorm:"not null" json:"userId"`
	ProjectTasks []ProjectTask `gorm:"constraint:OnDelete:CASCADE" json:"projectTasks"`
	Model
}

File: frontend/src/types/project.ts
export type ProjectTask = {
  id: string;
  name: string;
  status: "pending" | "in-progress" | "completed"; // extend if needed
};

export type Project = {
  id?: string;
  name: string;
  description?: string;
  userId?: string;
  projectTasks?: ProjectTask[];
  createdAt?: string;
  updatedAt?: string;
};
